scraper/src/Scraper.hs

```diff
@@ -22,7 +23,45 @@ runScrape = do
 scrape :: IO ()
 scrape = do 
   website <- httpLBS "http://leg.colorado.gov/bill-search?field_sessions=10171&sort_bef_combine=field_bill_number%20ASC" 
-  L8.writeFile "test.html" $ getResponseBody website
+  L8.writeFile "test/example/bill_page_1.html" $ getResponseBody website
+
+-- I'm very sorry. Forgive me for my sins.
+fixNextPage = fmap mconcat . fmap mconcat $ fmap fromMaybe' nextPage
+fixLastPage = fmap mconcat . fmap mconcat $ fmap fromMaybe' lastPage
+
+-- This is far from ideal. please change it later 
+fromMaybe' mval = 
+  case mval of 
+    Nothing  -> undefined 
+    Just val -> val
+
+nextPage :: IO (Maybe [[ByteString]]) 
+nextPage = do 
+  website <- readFile "test/example/bill_page_1.html"
+  return $ S.scrapeStringLike website next
+
+    where 
+      next :: S.Scraper ByteString [[ByteString]]
+      next = S.chroots ("li" S.@: [S.hasClass "pager-next"]) link
+
+      link :: S.Scraper ByteString [ByteString]
+      link = S.chroots "a" $ do 
+        href <- S.attr "href" S.anySelector
+        return href
+
+lastPage :: IO (Maybe [[ByteString]])
+lastPage = do 
+  website <- readFile "test/example/bill_page_1.html"
```

Comment:
Instead of reading the file, take the input as a parameter.

```haskell
lastPage :: ByteString -> IO (Maybe ByteString)
lastPage website = do ...
```

Then, you're not actually using any `IO` in the function if you're not reading the file, so your signature can be simplified to:

```haskell
lastPage :: ByteString -> Maybe ByteString
```

Reply:
This is complete.